### PR TITLE
Framework/spi

### PR DESCRIPTION
**Pull request by ATGr8**
Wrote SPI wrapper and .h file

### CODE REVIEW

**Comment by matthew-buglass** [comment]
Remember to add documentation to the C file.

**Comment by ATGr8** [comment]
> Remember to add documentation to the C file.

Thanks, I added them

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "files.associations": {
```
Comment (matthew-buglass) [inline .vscode/settings.json]: I don't think that the IDE setting should be uploaded to the repo

**Review by matthew-buglass — CHANGES_REQUESTED** [review]
Looks good, except for the .vscode settings being included. But I'm not sure what our policies on that, so I'll leave the to @ChrisAmaya to decide.

File: radsat-sk/framework/wrappers/RSPI.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ */
+
+#ifndef SPI_H_
+#define SPI_H_
+
+#include <hal/Drivers/SPI.h>
+#include <freertos/FreeRTOS.h>
+
+#define SPI_MAX_CS 8
+#define BUS 0
```
Comment (ChrisAmaya) [inline radsat-sk/framework/wrappers/RSPI.h:14]: you have 2 defines but they arent being used anywhere in your .c files

File: radsat-sk/framework/wrappers/RSPI.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+/***************************************************************************************************
+                                             FUNCTION DECLARATIONS
+***************************************************************************************************/
+
+uint32_t spiInit();
```
Comment (ChrisAmaya) [inline radsat-sk/framework/wrappers/RSPI.h]: You have no parameters in your function declaration but you have then in your .c

**Review by ChrisAmaya — CHANGES_REQUESTED** [review]

**Pull request closed without merging**